packages/react/src/components/Animation/Animation.tsx

```diff
@@ -77,9 +75,45 @@ export interface AnimationProps
    * - cubic-bezier(n,n,n,n) - Lets you define your own values in a cubic-bezier function
    */
   timingFunction?: string
+
+  /** Show the component; triggers the enter or exit animation. */
+  visible?: boolean
+
+  /** Run the enter animation when the component mounts, if it is initially shown. */
+  appear?: boolean
+
+  /** Wait until the first "enter" transition to mount the component (add it to the DOM). */
+  mountOnEnter?: boolean
+
+  /** Unmount the component (remove it from the DOM) when it is not shown. */
+  unmountOnExit?: boolean
+
+  /** The duration of the transition, in milliseconds. */
+  timeout?: number | { enter?: number; exit?: number; appear?: number }
+
+  /** Callback fired before the "entering" status is applied. An extra parameter isAppearing is supplied to indicate if the enter stage is occurring on the initial mount. */
+  onEnter?: (node: HTMLElement, isAppearing: boolean) => void
+
+  /** Callback fired after the "entering" status is applied. An extra parameter isAppearing is supplied to indicate if the enter stage is occurring on the initial mount. */
+  onEntering?: (node: HTMLElement, isAppearing: boolean) => void
+
+  /** Callback fired after the "entered" status is applied. An extra parameter isAppearing is supplied to indicate if the enter stage is occurring on the initial mount. */
+  onEntered?: (node: HTMLElement, isAppearing: boolean) => void
+
+  /** Callback fired before the "exiting" status is applied. */
+  onExit?: (node: HTMLElement) => void
```

Comment:
```suggestion
  onExit?: () => void
```

Do we want to expose `node` there? 🤔

Reply:
This is the API from the Transition component itself, I would rather not change anything for now...

Reply:
But maybe we should return here null as first argument for the event, and the node afterwards, to be consistent with the other callbacks. What do you think?

Maybe I am over-complicating, but let me know what you think @layershifter